libs/ui-components/src/components/form/validations.ts

```diff
@@ -306,28 +306,31 @@ export const validFleetRolloutPolicySchema = (t: TFunction) => {
       updateTimeout: Yup.number().required('Update timeout is required'),
       batches: Yup.array()
         .of(
-          Yup.object<BatchForm>().shape({
-            selector: validLabelsSchema(t),
-            limitType: Yup.string()
-              .oneOf([BatchLimitType.BatchLimitAbsoluteNumber, BatchLimitType.BatchLimitPercent])
-              .required(),
-            limit: Yup.number()
-              .test('correct-percentage', t('Percentage must be between 1 and 100.'), (num, { parent }) => {
-                if (num === undefined) {
+          Yup.lazy((values: BatchForm) =>
+            Yup.object<BatchForm>().shape({
+              selector: validLabelsSchema(t),
+              limitType: Yup.string()
+                .oneOf([BatchLimitType.BatchLimitAbsoluteNumber, BatchLimitType.BatchLimitPercent])
+                .required(),
+              limit: Yup.number()
+                .test('correct-percentage', t('Percentage must be between 1 and 100.'), (num) => {
+                  if (num === undefined) {
+                    return true;
+                  }
+                  if (values.limitType === BatchLimitType.BatchLimitPercent && num > 100) {
+                    return false;
+                  }
                   return true;
-                }
-                if ((parent as BatchForm).limitType === BatchLimitType.BatchLimitPercent && num > 100) {
-                  return false;
-                }
-                return true;
-              })
-              // When an absolute number, we can only restrict the minimum value
-              .min(1, t('Value must be greater or equal than 1')),
-            successThreshold: Yup.number()
-              .required('Success threshold is required')
-              .min(1, t('Success threshold must be between 1 and 100'))
-              .max(100, t('Success threshold must be between 1 and 100')),
-          }),
+                })
+                .test('not-decimal', t('Can not be decimal'), (val) => val === undefined || Number.isInteger(val))
```

Comment:
Small nit:

```suggestion
                .test('not-decimal', t('Cannot be decimal'), (val) => val === undefined || Number.isInteger(val))
```
sounds better?

Reply:
+1, updated